core/platform/Platform.cmake

```diff
@@ -8,6 +8,9 @@ elseif(${CMAKE_SYSTEM_NAME} STREQUAL "Windows")
     set(LF_PLATFORM_FILE lf_windows_support.c)
     set(CMAKE_SYSTEM_VERSION 10.0)
     message("Using Windows SDK version ${CMAKE_VS_WINDOWS_TARGET_PLATFORM_VERSION}")
+elseif(${CMAKE_SYSTEM_NAME} STREQUAL "Arduino")
+    set(LF_PLATFORM_FILE lf_arduino_support.cpp)
```

Comment:
This is a fix required by Arduino-CMake-Toolchain. Without the .cpp extension, the Arduino Build system will not link said library to Arduino's core library, which has CPP Files.

Reply:
Since it uses a mixture of different fixes that are a part of the C/CCpp targets, I instituted this middle ground solution. That said, it may be better later on to use the CCpp compiler for this purpose.